airflow/www/views.py

```diff
@@ -2894,9 +2894,13 @@ def action_muldelete(self, items):
 
     def process_form(self, form, is_created):
         """Process form data."""
-        formdata = form.data
-        if formdata['conn_type'] in ['jdbc', 'google_cloud_platform', 'grpc', 'yandexcloud', 'kubernetes']:
-            extra = {key: formdata[key] for key in self.extra_fields if key in formdata}
+        conn_type = form.data['conn_type']
+        extra = {
+            key: form.data[key]
+            for key in self.extra_fields
+            if key in form.data and key.startswith(f"extra__{conn_type}")
```

Comment:
Like in the `connection_form.js`, adding `__` at the end here avoids the (unlikely) case of substrings in the conn types, like so:
```Python
            if key in form.data and key.startswith(f"extra__{conn_type}__")
```

Reply:
Oh yeah! Good call :)